packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/resource_usage_tracker/service_runs.py

```diff
@@ -54,6 +59,36 @@ async def get_service_run_page(
     return result
 
 
+@log_decorator(_logger, level=logging.DEBUG)
+async def get_osparc_credits_aggregated_usages_page(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    user_id: UserID,
+    product_name: ProductName,
+    aggregated_by: ServicesAggregatedUsagesType,
+    time_period: ServicesAggregatedUsagesTimePeriod,
```

Comment:
why is this an enum and not just a datetime.timedelta?

Reply:
In reality, it is an enum. I do not want to give too much flexibility to the client, because I foresee that the operation might be costly to compute on the fly and therefore we will need to precompute and store the data somewhere. For now, the frontend will give only these 3 options to the user.